src/dao/user.dao.ts

```diff
@@ -7,32 +7,60 @@ import { formSchema } from "@/shared/onboarding";
 
 class UserDAO {
   async createUser(loginUser: NextUser) {
-    const existingUser = await this.fetchUserBy({
-      email: loginUser.email ?? raise("E-mail not found"),
-    });
+    if (!loginUser.email) {
+      throw raise("E-mail not found");
+    }
+    const existingUser = await this.fetchUserBy({ email: loginUser.email });
 
     if (existingUser) return existingUser;
 
-    return await prisma.user.create({
+    return prisma.user.create({
       data: this.parseNextAuthUser(loginUser),
     });
   }
 
   async fetchUserBy(where: Prisma.UserWhereUniqueInput) {
-    if (where.username) {
-      where.username = decodeURIComponent(where.username);
+    const query = { ...where };
+    if (query.username) {
+      query.username = decodeURIComponent(query.username);
     }
 
-    return await prisma.user.findUnique({
-      where,
-    });
+    return prisma.user.findUnique({ where: query });
   }
 
-  async isOboardingFinished(username: string) {
-    const user =
-      (await this.fetchUserBy({ username })) ??
-      raise("Could not find user on DB");
+  async fetchUsers(
+    filter: { username?: string; email?: string },
+    page = 1,
+    pageSize = 10
+  ) {
+    const skip = (page - 1) * pageSize;
+    const where: Prisma.UserWhereInput = {};
 
+    if (filter.email || filter.username) {
```

Comment:
Writing this again, I will show you today in our meeting. Leaving this for reference:
https://www.prisma.io/docs/orm/prisma-client/special-fields-and-types/null-and-undefined